### DISCUSSION TITLE
Are conditional parameters in the configuration space?

### DISCUSSION THREAD

**Discussion by AndreBiedenkapp** [ds4dm/ml4co-competition#28] [Q&A]
Hi,

looking at the [list of parameters for the configuration task](https://www.scipopt.org/doc/html/PARAMETERS.php) we are wondering if some of the parameters are conditional. I.e. do some parameters only have effects if another parameter is set to a particular value?
As far as I could see only forbidden parameters are given https://github.com/ds4dm/ml4co-competition/blob/666b5d941a9ac6f91897447893a2c9c936ec7539/common/environments.py#L116-L122

Best,
André

**Comment by mlubin** [comment]
Yes, some parameters are conditional. For example, `separating/clique/freq` can be used to disable the clique separator. When it's disabled, then the other `separating/clique/...` parameters likely have no effect. There are many such parameters in SCIP.

**Reply by AndreBiedenkapp** [reply]
Thank you for the answer :)

Are all conditional parameters identifiable by the name?
If not, is there a list/other way of finding conditional effects?

**Reply by simonbowly** [reply]
The main parameter names to look for are:
* setting `freq` parameters to -1 disables the relevant separator, propagator, or heuristic
* setting `maxrounds` or `maxroundsroot` to 0 similarly disables separators or propagators
* setting `maxdepth` to 0 for separators, heuristics, or branching rules prevents their use after the root node
* setting `priority` may have a conditional effect as it may result in branching rules, node selectors, etc not being run if higher priority rules succeed
